spring-pulsar/src/main/java/org/springframework/pulsar/core/DefaultSchemaResolver.java

```diff
@@ -129,8 +132,22 @@ public <T> Schema<T> getSchema(Class<?> messageClass, boolean returnDefault) {
 	}
 
 	@Nullable
-	private Schema<?> getCustomSchemaOrMaybeDefault(Class<?> messageClass, boolean returnDefault) {
-		return this.customSchemaMappings.getOrDefault(messageClass, (returnDefault ? Schema.BYTES : null));
+	protected Schema<?> getCustomSchemaOrMaybeDefault(Class<?> messageClass, boolean returnDefault) {
+		Schema<?> schema = this.customSchemaMappings.get(messageClass);
```

Comment:
I wonder if we should make `customSchemaMappings` protected here so that custom impls could use it as well?

Reply:
Child classes can already add/remove/getView. I'm not sure what else could be needed. Maybe this could be studied in a subsequent issue/PR ?

Reply:
Yep, that should do. And good point - lets leave the "As a user I can easily extend the topic and schema resolver to suit my needs" story for another issue. On that note, please resolve (table) any of my related comments around extension on this and the other resolver PR.

Reply:
https://github.com/spring-projects-experimental/spring-pulsar/issues/330